### ISSUE TITLE
New cop: RUBY_VERSION in gemspec

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#7137]
## Is your feature request related to a problem? Please describe.

Gemspec should not contain RUBY_VERSION as a condition to switch dependencies.
For example:

```ruby
Gem::Specification.new do |s|
  if RUBY_VERSION >= '2.5'
    s.add_runtime_dependency 'gem_a'
  else
    s.add_runtime_dependency 'gem_b'
  end
end
```

It has unexpected behavior. The `RUBY_VERSION` is determined by `rake release`.
If I use Ruby 2.5 to execute `rake release`, the gem always has `gem_a` as a dependency even if a user installs the gem with Ruby 2.4 or older.

We have several ways to fix the problem.

* Remove the dependencies and use post_install_message to notice the gems.
* Add both gems to the dependency.
* If the dependencies are for development, move them to Gemfile.

## Describe the solution you'd like

Detect reference of `RUBY_VERSION` in `*.gemspec` and add an offense to it.
For example: `def_node_matcher :ruby_version?, '(const nil :RUBY_VERSION)'`

## Describe alternatives you've considered

The solution is loose. It checks `RUBY_VERSION` in anywhere of gemspec.
We can implement stricter detection. For example, detect `RUBY_VERSION` only if it is placed a conditon.

But I think the looser version is better. Because I guess the cop will have many false negative if we choose the stricter version. And I guess the looser version does not have too many false positives.

## Additional context

Today I found using RUBY_VERSION in a gemspec.
https://github.com/rafaelsales/ulid/pull/17
So I think the cop is meaningful.

**Comment by bbatsov** [comment]
You've got my blessing! :-)

I was also thinking we should add a section on Gemfile/gemspec to https://rubystyle.guide/. Our cops have a lot of good advice that's not part of the guide yet.

**Comment by pocke** [comment]
> I was also thinking we should add a section on Gemfile/gemspec to https://rubystyle.guide/. Our cops have a lot of good advice that's not part of the guide yet.


Definitely! It will be helpful for beginners of gem developer.
